compliance/collection/compliance/compliance.go

```diff
@@ -91,6 +93,11 @@ func (c *Compliance) Start() {
 		}
 	}
 
+	if env.NodeScanningV4Enabled.BooleanSetting() {
```

Comment:
```suggestion
	if env.NodeScanningV4Enabled.BooleanSetting() && c. nodeIndexer != nil {
```
Would that make sense or it is not necessary because of `nodeIndexer` being the interface?

Reply:
Can you pass nil to the constructor?